### DISCUSSION TITLE
Can webassembly/javascript be added?

### DISCUSSION THREAD

**Discussion by gautam1168** [compiler-explorer/compiler-explorer#4700] [Ideas]
Hello,

the `d8` executable that is created when compiling `v8` allows one to print out the bytecode for a javascript program by running
```
d8.exe javascript.js --print-bytecode
```
It also allows flags like `--print-code`, `--print-opt-code` and `--print-wasm-code` to output the asm code generated from a javascript or webassembly input. Can godbolt support showing these outputs for javascript/wasm?

**Comment by partouf** [comment]
we have an old issue about this @ https://github.com/compiler-explorer/compiler-explorer/issues/264
please feel free to submit a pullrequest to support it

**Comment by gauravgautamgoldcast** [comment]
Thankyou, Ill get started. I guess I need to first do a local setup before I can attempt to add the feature. I already have a local compilation of v8 so I can generate all the assembly and bytecode. (oh sorry I posted from my work account here)

**Reply by AbrilRBS** [reply]
No worries. Let us know if you have any problems, questions or issues, we'll be happy to help :)

**Comment by gautam1168** [comment]
So I now have a local setup of the app where I am able to generate the bytecode and asm for javascript as generated by v8. Bytecode is on the right and its on stdout. 

![image](https://user-images.githubusercontent.com/9499740/218253433-390631fa-ecf8-4a88-955f-73327df3fa54.png)

**Comment by gautam1168** [comment]
Opened a PR: https://github.com/compiler-explorer/compiler-explorer/pull/4721

**Comment by gautam1168** [comment]
d8 has a lot of flags to trace various things about compilation and execution. For example look for the word trace in this file: https://github.com/v8/v8/blob/master/src/flags/flag-definitions.h

I think there should be some way to see the output of these in the compiler explorer. Would it be possible to add some features to show these? One example can be found in this web.dev blogpost where some tracing is done for optimizing some javascript. https://web.dev/performance-mystery/

From the above blogpost the visualization of the trace:
![image](https://user-images.githubusercontent.com/9499740/218296421-cc37aab6-2ed4-460f-9519-ab7bb707b574.png)

I'm thinking there should be a way to watch these in the explorer as well. Would this be out of scope of compiler explorer? Since v8 is JITed, some real value can be gained from building the ability to see these traces easily. I don't think chrome dev tools allows us to see these things, so its a unique thing that is hard to do unless one goes and builds v8 on their own.

**Reply by AbrilRBS** [reply]
These would not be the first tools added specifically for only one compiler, so as far as I'm concerned, this would be cool to have, but @mattgodbolt might have other opinions. The issue comes from actually integrating such tools in CE, which might be a bit tricky, so let us know if you need any help with that :)

**Reply by gautam1168** [reply]
I will. Thankyou for replying.
